src/gui/guiFormSpecMenu.cpp

```diff
@@ -4021,11 +4025,16 @@ bool GUIFormSpecMenu::OnEvent(const SEvent& event)
 {
 	if (event.EventType==EET_KEY_INPUT_EVENT) {
 		KeyPress kp(event.KeyInput);
-		if (event.KeyInput.PressedDown && (
-				(kp == EscapeKey) ||
-				((m_client != NULL) && (kp == getKeySetting("keymap_inventory"))))) {
-			tryClose();
-			return true;
+		if (event.KeyInput.PressedDown) {
+			if (kp == EscapeKey && event.KeyInput.Control) {
+				m_text_dst->gotText(L"MenuQuit");
```

Comment:
What is this needed for? It seems to work just as well without this.

Reply:
This allows Luanti to be closed at the main menu when Ctrl + Escape is pressed. Without it, Escape would close the window but Ctrl + Escape would not.

Reply:
hmm, i wouldn't consider it necessary for Ctrl + Escape to close the window, but i guess it makes sense if you think of Ctrl + Escape as "stronger" than just Escape